pkg/rollup/resolution.go

```diff
@@ -0,0 +1,182 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package rollup
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/pgxconn"
+)
+
+const (
+	originalSchema          = "prom_data"
+	upperLimit              = 5000 // Maximum samples allowed
+	assumedScrapeInterval   = time.Second * 30
+	refreshRollupResolution = time.Minute * 30
+)
+
+type rollupInfo struct {
+	schemaName string
+	resolution time.Duration
+}
+
+type QueryHelper struct {
+	conn                pgxconn.PgxConn
+	ctx                 context.Context
+	refreshMtx          sync.RWMutex
+	metricType          map[string]string // metric_name: metric_type
+	resolutionsASC      []rollupInfo      // {schemaName, resolution} in ascending order of resolution.
```

Comment:
Since this is just an array of rollups, why not name it `rollups`? I also feel that we don't strictly need to make ordering part of a name (at least the most of our code doesn't seem to follow that practice)